src/components/MyAccount/NewSettings/SettingsSelectForm.tsx

```diff
@@ -0,0 +1,199 @@
+import { useContext, useState } from "react"
+import { PatronDataContext } from "../../../context/PatronDataContext"
+import {
+  Button,
+  Flex,
+  Icon,
+  Select,
+  SkeletonLoader,
+  Text,
+} from "@nypl/design-system-react-components"
+import SettingsLabel from "./SettingsLabel"
+import SaveCancelButtons from "./SaveCancelButtons"
+import type { Patron, SierraCodeName } from "../../../types/myAccountTypes"
+import EditButton from "./EditButton"
+
+interface HomeLibraryNotificationFormProps {
+  type: "library" | "notification"
+  patronData: Patron
+  settingsState
+  pickupLocations: SierraCodeName[]
+}
+
+const HomeLibraryNotificationForm = ({
+  type,
+  patronData,
+  settingsState,
+  pickupLocations,
+}: HomeLibraryNotificationFormProps) => {
+  const { getMostUpdatedSierraAccountData } = useContext(PatronDataContext)
+  const [isLoading, setIsLoading] = useState(false)
+  const [isEditing, setIsEditing] = useState(false)
+  const [error, setError] = useState(false)
+
+  const { setIsSuccess, setIsFailure, isOtherEditing, setIsOtherEditing } =
+    settingsState
+
+  const notificationPreferenceMap = [
+    { code: "z", name: "Email" },
+    { code: "p", name: "Phone" },
+    { code: "-", name: "None" },
+  ]
+
+  const validateInput = (input) => {
+    if (input == "Phone" && patronData.phones.length === 0) {
+      setError(true)
+    }
+  }
+
+  const sortedPickupLocations = [
+    patronData.homeLibrary,
+    ...pickupLocations.filter(
+      (loc) => loc.code.trim() !== patronData.homeLibrary.code.trim()
+    ),
+  ]
+
+  const options =
+    type === "notification" ? notificationPreferenceMap : sortedPickupLocations
```

Comment:
can this be added to the form utils body?
also note that we don't want to allow patrons to change their preference to none, so this option should be removed from the dropdown if it's not their currently selected option.

Reply:
also note that we don't want to allow patrons to change their preference to none, so this option should be removed from the dropdown if it's not their currently selected option.

Reply:
Had to add none as a possibility to the Patron type to do this, not sure why that wasn't already the case, but lmk if that's intentional.

Reply:
Hmm I imagine that it has something to do with not letting patrons sign up without an email or phone number. Probably ok but let's make sure to flag it for QA.